DH_Engine/Classes/DHVotingHandler.uc

```diff
@@ -58,6 +58,87 @@ function Timer()
     }
 }
 
+// Modified to handle map specific repeat limit
+function AddMap(string MapName, string Mutators, string GameOptions)
+{
+    local MapHistoryInfo        MapInfo;
+    local bool                  bUpdate;
+    local int                   i, MapRepeatLimit;
+    local JSONObject            MapObject;
+    local string                DecodedString;
+
+    // Dont add duplicate map names
+    for (i = 0; i < MapList.length; i++)
+    {
+        if(MapName ~= MapList[i].MapName)
+        {
+            return;
+        }
+    }
+
+    MapInfo = History.GetMapHistory(MapName);
+
+    MapList.length = MapCount + 1;
+
+    if (MapInfo.G != "")
+    {
+        DecodedString = class'UTF8Encoding'.static.FromByteArray(class'Base64Encoding'.static.Decode(MapInfo.G));
+
+        MapObject = (new class'JSONParser').ParseObject(DecodedString);
+
+        if (MapObject != none)
+        {
+            MapObject.PutString("MapName", MapName);
+            MapList[MapCount].MapName = MapObject.Encode();
+        }
+        else
+        {
+            Warn("MapObject failed to parse in DHVotingHandler for:" @ MapName);
+            MapList[MapCount].MapName = MapName;
+        }
+    }
+    else
+    {
+        MapList[MapCount].MapName = MapName;
+    }
+
+    MapList[MapCount].PlayCount = MapInfo.P;
+    MapList[MapCount].Sequence = MapInfo.S;
+
+    // Check if there is specific repeat limit for this map
+    if (MapObject != none && MapObject.Get("RepeatLimit").AsString() != "")
```

Comment:
You can simply check `MapObject.Get("RepeatLimit") != none` here to achieve the same effect.